<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8" /><meta name="generator" content="Docutils 0.18.1: http://docutils.sourceforge.net/" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>Step 4. Analysis &mdash; STEGO.R 0.1 documentation</title>
      <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
      <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
      <link rel="stylesheet" href="_static/custom.css" type="text/css" />
  <!--[if lt IE 9]>
    <script src="_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
        <script src="_static/jquery.js?v=5d32c60e"></script>
        <script src="_static/_sphinx_javascript_frameworks_compat.js?v=2cd50e6c"></script>
        <script data-url_root="./" id="documentation_options" src="_static/documentation_options.js?v=2709fde1"></script>
        <script src="_static/doctools.js?v=888ff710"></script>
        <script src="_static/sphinx_highlight.js?v=dc90522c"></script>
    <script src="_static/js/theme.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="User Tips for the analysis" href="5.UserTips.html" />
    <link rel="prev" title="Step 3. Quality control" href="3.Analysis.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >

          
          
          <a href="index.html">
            
              <img src="_static/logo-removebg-preview_green.png" class="logo" alt="Logo"/>
          </a>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <ul class="current">
<li class="toctree-l1"><a class="reference internal" href="0.Installation.html">Overview</a></li>
<li class="toctree-l1"><a class="reference internal" href="1.Backend.html">Step 1. Formatting</a></li>
<li class="toctree-l1"><a class="reference internal" href="2.Engine.html">Step 2. TCRex and ClusTCR2</a></li>
<li class="toctree-l1"><a class="reference internal" href="3.Analysis.html">Step 3. Quality control</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">Step 4. Analysis</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#application-structure">Application structure</a></li>
<li class="toctree-l2"><a class="reference internal" href="#uploaded-files">Uploaded files</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#side-bar">Side bar</a></li>
<li class="toctree-l3"><a class="reference internal" href="#main-panel">Main panel</a></li>
<li class="toctree-l3"><a class="reference internal" href="#side-bar-analysis-sections">Side bar (Analysis sections)</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#overview">Overview</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#side-bar-ov-only">Side bar (OV only)</a></li>
<li class="toctree-l3"><a class="reference internal" href="#main-panel-ov">Main panel (OV)</a><ul>
<li class="toctree-l4"><a class="reference internal" href="#gex">GEx</a></li>
<li class="toctree-l4"><a class="reference internal" href="#tcr">TCR</a></li>
</ul>
</li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#tcr-gex">TCR -&gt; GEx</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#sidebar-common-features">Sidebar common features</a></li>
<li class="toctree-l3"><a class="reference internal" href="#clonal-abundance">Clonal abundance</a></li>
<li class="toctree-l3"><a class="reference internal" href="#expansion">Expansion</a></li>
<li class="toctree-l3"><a class="reference internal" href="#clustcr2">ClusTCR2</a></li>
<li class="toctree-l3"><a class="reference internal" href="#epitope-tcrex">Epitope (TCRex)</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#automation-tcr-gex">Automation (TCR -&gt; GEX)</a></li>
<li class="toctree-l2"><a class="reference internal" href="#gex-tcr">GEx -&gt; TCR</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="5.UserTips.html">User Tips for the analysis</a></li>
<li class="toctree-l1"><a class="reference internal" href="6.FAQ.html">Frequenctly Asked Questions</a></li>
<li class="toctree-l1"><a class="reference internal" href="7.Tutorial.html">Video Tutorial</a></li>
<li class="toctree-l1"><a class="reference internal" href="8.History.html">History</a></li>
<li class="toctree-l1"><a class="reference internal" href="9.Publications.html">Publications and citations</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="index.html">STEGO.R</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="index.html" class="icon icon-home" aria-label="Home"></a></li>
      <li class="breadcrumb-item active">Step 4. Analysis</li>
      <li class="wy-breadcrumbs-aside">
            <a href="_sources/4.ExperimentalDesign.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="step-4-analysis">
<h1>Step 4. Analysis<a class="headerlink" href="#step-4-analysis" title="Permalink to this heading"></a></h1>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>This section goes over what is present in STEGO.R. The video tutorial will present some examples of each that will depend on specific biological questions.</p>
</div>
<section id="application-structure">
<h2>Application structure<a class="headerlink" href="#application-structure" title="Permalink to this heading"></a></h2>
<p>The analysis section is divided into several sub-sections.</p>
<ul class="simple">
<li><p>Uploaded files</p></li>
<li><p>Overview</p></li>
<li><p>TCR -&gt; GEX</p></li>
<li><p>Automation (TCR -&gt; GEx)</p></li>
<li><p>GEx -&gt; TCR</p></li>
</ul>
</section>
<section id="uploaded-files">
<h2>Uploaded files<a class="headerlink" href="#uploaded-files" title="Permalink to this heading"></a></h2>
<p>There first tab will show in the side panel the files that can be uploaded.</p>
<section id="side-bar">
<h3>Side bar<a class="headerlink" href="#side-bar" title="Permalink to this heading"></a></h3>
<p>To streamline the analysis, for this section you will only see the following:</p>
<a class="reference internal image-reference" href="_images/4_files_to_upload.png"><img alt="Alternative text" src="_images/4_files_to_upload.png" style="width: 400px;" /></a>
<p>This will include the following default files that can be uploaded:
- Upload seurat file (.rds)
- Upload AG clusTCR2 file (.csv)
- Upload BD clusTCR2 file (.csv)
- Upload TCRex (.tsv)</p>
<p>To make selecting the “<strong>Data source</strong>” and “<strong>Species</strong>”, there are functions to identify if the TCR-seq formattting is AIRR or not. The species is currently either human (hs) or mouse (mm) and based on the gene formatting of all caps (hs) or proper case (mm). The latter is needed as R is a case sensative language.</p>
<p>If you need additional identifiers for the analysis section, this can be added by switching “add additional labels” to “yes” and upload the “Update_labels.csv”. The file consists of the first column labelled as “ID” must not be changed. This first column must match the unique identifiers used in the “Sample_Name” column for ease of use. The user can add as many columns. Any special characters will be convered to either .  or _. An example is given with the colitis dataset that has three broad categories under the condition column. The user can then add additional variables that could be used for analysis.This was added to the analysis section, so the user does not have to go back to Step 1, and is also restrictive to one unique label.</p>
<a class="reference internal image-reference" href="_images/4_updating_labs.png"><img alt="Alternative text" src="_images/4_updating_labs.png" style="width: 300px;" /></a>
<table class="longtable docutils align-default" id="id1">
<caption><span class="caption-text">Updated labelling for the colitis dataset.</span><a class="headerlink" href="#id1" title="Permalink to this table"></a></caption>
<colgroup>
<col style="width: 40%" />
<col style="width: 60%" />
</colgroup>
<thead>
<tr class="row-odd"><th class="head"><p>ID</p></th>
<th class="head"><p>Condition</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>CD3_C1</p></td>
<td><p>colitis</p></td>
</tr>
<tr class="row-odd"><td><p>CD3_C2</p></td>
<td><p>colitis</p></td>
</tr>
<tr class="row-even"><td><p>CD3_C3</p></td>
<td><p>colitis</p></td>
</tr>
<tr class="row-odd"><td><p>CD3_C4</p></td>
<td><p>colitis</p></td>
</tr>
<tr class="row-even"><td><p>CD3_C5</p></td>
<td><p>colitis</p></td>
</tr>
<tr class="row-odd"><td><p>CD3_C6</p></td>
<td><p>colitis</p></td>
</tr>
<tr class="row-even"><td><p>CD3_C7</p></td>
<td><p>colitis</p></td>
</tr>
<tr class="row-odd"><td><p>CD3_C8</p></td>
<td><p>colitis</p></td>
</tr>
<tr class="row-even"><td><p>CD3_CT1</p></td>
<td><p>normal control</p></td>
</tr>
<tr class="row-odd"><td><p>CD3_CT2</p></td>
<td><p>normal control</p></td>
</tr>
<tr class="row-even"><td><p>CD3_CT3</p></td>
<td><p>normal control</p></td>
</tr>
<tr class="row-odd"><td><p>CD3_CT4</p></td>
<td><p>normal control</p></td>
</tr>
<tr class="row-even"><td><p>CD3_CT5</p></td>
<td><p>normal control</p></td>
</tr>
<tr class="row-odd"><td><p>CD3_CT6</p></td>
<td><p>normal control</p></td>
</tr>
<tr class="row-even"><td><p>CD3_CT7</p></td>
<td><p>normal control</p></td>
</tr>
<tr class="row-odd"><td><p>CD3_CT8</p></td>
<td><p>normal control</p></td>
</tr>
<tr class="row-even"><td><p>CD3_NC1</p></td>
<td><p>no colitis</p></td>
</tr>
<tr class="row-odd"><td><p>CD3_NC2</p></td>
<td><p>no colitis</p></td>
</tr>
<tr class="row-even"><td><p>CD3_NC3</p></td>
<td><p>no colitis</p></td>
</tr>
<tr class="row-odd"><td><p>CD3_NC4</p></td>
<td><p>no colitis</p></td>
</tr>
<tr class="row-even"><td><p>CD3_NC5</p></td>
<td><p>no colitis</p></td>
</tr>
<tr class="row-odd"><td><p>CD3_NC6</p></td>
<td><p>no colitis</p></td>
</tr>
<tr class="row-even"><td><p>CD3_NC7</p></td>
<td><p>no colitis</p></td>
</tr>
<tr class="row-odd"><td><p>CD3_NC8</p></td>
<td><p>no colitis</p></td>
</tr>
</tbody>
</table>
</section>
<section id="main-panel">
<h3>Main panel<a class="headerlink" href="#main-panel" title="Permalink to this heading"></a></h3>
<p>Depending on what you have uploaded, the main panel will render several tables. The first table is of the Seurat object’s meta-data and you will see how it adds the additional labels after the “Sample_Name” column.</p>
<p>Uploaded example of the colitis complication to melanoma therapy dataset (GSE144469)</p>
<a class="reference internal image-reference" href="_images/4_file_uploading.png"><img alt="Alternative text" src="_images/4_file_uploading.png" style="width: 1200px;" /></a>
</section>
<section id="side-bar-analysis-sections">
<h3>Side bar (Analysis sections)<a class="headerlink" href="#side-bar-analysis-sections" title="Permalink to this heading"></a></h3>
<p>The side bar has variables that may be relevant to the all analyses, or only a specific section.</p>
<p>The variables common to the analysis section include:</p>
<ul class="simple">
<li><p>V gene with/without CDR3</p></li>
<li><p>Selected Sample</p></li>
<li><dl class="simple">
<dt>Restricing to one sample for filtering purposes</dt><dd><ul>
<li><p>The purpose of the restrict to one sample can be to only show one individual in the overview section. Under the clonal abudance, this will change the summary table to represent only the individual of interest.</p></li>
</ul>
</dd>
</dl>
</li>
<li><p>Colouring</p></li>
<li><p>Plot text</p></li>
<li><p>Legend</p></li>
</ul>
<a class="reference internal image-reference" href="_images/4_ALL_general_features.png"><img alt="Alternative text" src="_images/4_ALL_general_features.png" style="width: 300px;" /></a>
<p>Within each of the colouring, plot text and legend.</p>
<a class="reference internal image-reference" href="_images/4_ALL_colouring.png"><img alt="Alternative text" src="_images/4_ALL_colouring.png" style="width: 300px;" /></a>
<a class="reference internal image-reference" href="_images/4_ALL_plot_parameter.png"><img alt="Alternative text" src="_images/4_ALL_plot_parameter.png" style="width: 300px;" /></a>
<a class="reference internal image-reference" href="_images/4_ALL_legend.png"><img alt="Alternative text" src="_images/4_ALL_legend.png" style="width: 300px;" /></a>
</section>
</section>
<section id="overview">
<h2>Overview<a class="headerlink" href="#overview" title="Permalink to this heading"></a></h2>
<p>The purpose of this section was to be able to interrogate the overview of the GEx only population dynamics regardless of TCR-seq and visa versa with TCR-seq only.</p>
<section id="side-bar-ov-only">
<h3>Side bar (OV only)<a class="headerlink" href="#side-bar-ov-only" title="Permalink to this heading"></a></h3>
<p>There are two aspects that are specific to the overview section that includes splitting the graphs based on the “selected sample”, adn in the TCR section for colouring by the binned clone counts, clone frequency or top expanded clones.</p>
<a class="reference internal image-reference" href="_images/4_OV_split.png"><img alt="Alternative text" src="_images/4_OV_split.png" style="width: 300px;" /></a>
<a class="reference internal image-reference" href="_images/4_OV_type_of_graph.png"><img alt="Alternative text" src="_images/4_OV_type_of_graph.png" style="width: 300px;" /></a>
</section>
<section id="main-panel-ov">
<h3>Main panel (OV)<a class="headerlink" href="#main-panel-ov" title="Permalink to this heading"></a></h3>
<p>The GEx section contains the following sections: GEx and TCR</p>
<section id="gex">
<h4>GEx<a class="headerlink" href="#gex" title="Permalink to this heading"></a></h4>
<p>This section focuses on a general interrogation of the annotations regardless of TCR</p>
<ul class="simple">
<li><dl class="simple">
<dt>Percentage</dt><dd><ul>
<li><p>This section is divided into percentage (Selected sample vs Colour by:) and can be downloaded as a .csv file.</p></li>
</ul>
</dd>
</dl>
</li>
<li><dl class="simple">
<dt>UMAP plot</dt><dd><ul>
<li><p>This can be have the full dataset or split by selected samples.</p></li>
<li><p>Samples can be removed from the “Order of graph (Selected Sample)”</p></li>
<li><p>Additionally, as there can be overlapping with labels, the user can chose in the “Show all labels?” to show all or selected.</p></li>
</ul>
</dd>
</dl>
</li>
<li><dl class="simple">
<dt>Summary chart</dt><dd><ul>
<li><p>As we used the semi-supervised annotation stratergy, this usually causes lots of overlap in the UMAP plot.</p></li>
<li><p>The user can display the “Colour by:” as any of the variables as either a pie chart or bar plot.</p></li>
</ul>
</dd>
</dl>
</li>
</ul>
<a class="reference internal image-reference" href="_images/4_OV_UMAP_split.png"><img alt="Alternative text" src="_images/4_OV_UMAP_split.png" style="width: 1200px;" /></a>
</section>
<section id="tcr">
<h4>TCR<a class="headerlink" href="#tcr" title="Permalink to this heading"></a></h4>
<p>This section focuses on TCR expansion, and include replicating features from scRepertoire i.e., clonal expansion presented on the UMAP.</p>
<ul class="simple">
<li><dl class="simple">
<dt>Overlap</dt><dd><ul>
<li><p>Summary table to show the clonal expansion based on the Selected Sample column</p></li>
<li><p>Upset table that is “V gene with/without CDR3” vs “Selected Sample”</p></li>
<li><p>If &lt;31 groups to compare, it will render an UPSET plot. This is a type of VENN diagram for more complex comparisons. The top bar represents the unique clones that overlapped the samples if there was a dot and line connection. The right bar graph is the unique clones for each sample</p></li>
</ul>
</dd>
</dl>
</li>
<li><dl class="simple">
<dt>Line Graph</dt><dd><ul>
<li><p>This section was developed for explore multi-sample/time series based data</p></li>
<li><p>The way the samples were labelled can split the text by - . etc.</p></li>
</ul>
</dd>
</dl>
</li>
<li><dl class="simple">
<dt>Clonal expansion plot</dt><dd><ul>
<li><p>bar plot of the clonal</p></li>
</ul>
</dd>
</dl>
</li>
<li><dl class="simple">
<dt>Clonality (counts)</dt><dd><ul>
<li><p>UMAP plot</p></li>
</ul>
</dd>
</dl>
</li>
</ul>
<p><em>Note: Clonal expansion plot and  Clonality (counts) variable can be changed with counts (Number_expanded), frequency (Frequency_Expanded) or most abundant clones (Top_clonotypes).</em></p>
<a class="reference internal image-reference" href="_images/4_OV_TCRonly.png"><img alt="Alternative text" src="_images/4_OV_TCRonly.png" style="width: 300px;" /></a>
</section>
</section>
</section>
<section id="tcr-gex">
<h2>TCR -&gt; GEx<a class="headerlink" href="#tcr-gex" title="Permalink to this heading"></a></h2>
<p>This section is split into four subsections.</p>
<ul class="simple">
<li><p>Clonal abudance</p></li>
<li><p>Exapanded</p></li>
<li><p>ClusTCR</p></li>
<li><p>Epitope</p></li>
</ul>
<section id="sidebar-common-features">
<h3>Sidebar common features<a class="headerlink" href="#sidebar-common-features" title="Permalink to this heading"></a></h3>
<a class="reference internal image-reference" href="_images/4_TCRGEX_collapsed.png"><img alt="Alternative text" src="_images/4_TCRGEX_collapsed.png" style="width: 300px;" /></a>
<ul class="simple">
<li><dl class="simple">
<dt>Heatmap</dt><dd><ul>
<li><p>The user can change the zero, from one count, maximum count colouring scale.</p></li>
<li><p>This is based on total counts, and I have not yet developed a frequency scaling.</p></li>
</ul>
</dd>
</dl>
</li>
<li><dl class="simple">
<dt>Cut-offs for FindMarker (Defaults)</dt><dd><ul>
<li><p>logFC = 0.25</p></li>
<li><p>Minimum point count = 0.25</p></li>
<li><p>adjusted p-value = 0.1</p></li>
<li><p>Bins for the dot plot (changed from seruat default 25 to 10).</p></li>
</ul>
</dd>
</dl>
</li>
<li><dl class="simple">
<dt>Dot plot</dt><dd><ul>
<li><p>Change the colours of the normalised expression from lower, mid and high.</p></li>
<li><p>Also the dotplot will by default only display the 30 most significant transcripts. This was chosen based on developer preference of high quality figures with the default text sizes.</p></li>
</ul>
</dd>
</dl>
</li>
<li><dl class="simple">
<dt>Violin plot</dt><dd><ul>
<li><p>The user can keep in the point (jitter)</p></li>
<li><p>The user can alter the transparancy of the violin plot. This is needed as the jittered dot points are behind the violin layer in the graph.</p></li>
</ul>
</dd>
</dl>
</li>
</ul>
<a class="reference internal image-reference" href="_images/4_TCRGEX_heatmap.png"><img alt="Alternative text" src="_images/4_TCRGEX_heatmap.png" style="width: 300px;" /></a>
<a class="reference internal image-reference" href="_images/4_TCRGEX_cutoff.png"><img alt="Alternative text" src="_images/4_TCRGEX_cutoff.png" style="width: 300px;" /></a>
<a class="reference internal image-reference" href="_images/4_TCRGEX_dotplot.png"><img alt="Alternative text" src="_images/4_TCRGEX_dotplot.png" style="width: 300px;" /></a>
<a class="reference internal image-reference" href="_images/4_TCRGEX_violin.png"><img alt="Alternative text" src="_images/4_TCRGEX_violin.png" style="width: 300px;" /></a>
</section>
<section id="clonal-abundance">
<h3>Clonal abundance<a class="headerlink" href="#clonal-abundance" title="Permalink to this heading"></a></h3>
<p>The purpose of the clonal abundance section is for the interrogation of single clones or groups of clones.</p>
<p>The user can select the type of comparison to:</p>
<ul class="simple">
<li><dl class="simple">
<dt>Background (BG)</dt><dd><ul>
<li><p>This will compare the selected clone(s) to the rest of the data provided.</p></li>
</ul>
</dd>
</dl>
</li>
<li><dl class="simple">
<dt>singlets</dt><dd><ul>
<li><p>This will compare the selected clone(s) to the cells with a TCR based on the “V gene with/without CDR3” that were only observed once in the data, as these are less likely to be antigen specific.</p></li>
</ul>
</dd>
</dl>
</li>
<li><dl class="simple">
<dt>Clones</dt><dd><ul>
<li><p>compair to a specific clone(s). This will bring up the ident.2 to select the clones.</p></li>
</ul>
</dd>
</dl>
</li>
</ul>
<p>In order to render the table efficiently, I have included a filter under the clonal abundance section to the top 50, as well as included a minimum count threshold (default = 2).</p>
<a class="reference internal image-reference" href="_images/4_TCRGEX_abundance.png"><img alt="Alternative text" src="_images/4_TCRGEX_abundance.png" style="width: 300px;" /></a>
<a class="reference internal image-reference" href="_images/4_TCRGEX_clonal_abudance_section.png"><img alt="Alternative text" src="_images/4_TCRGEX_clonal_abudance_section.png" style="width: 1200px;" /></a>
</section>
<section id="expansion">
<h3>Expansion<a class="headerlink" href="#expansion" title="Permalink to this heading"></a></h3>
<p>This section was aimed to interrogate if there was a consistant gene expression signature of the expanded clones regarless of the TCR arragement.</p>
<p>The sidebar contains some variables to help decide on what the count cut-off could be.</p>
<p>The “Cut off greater than” helps calculate by defaul the D50, or the count above which 50% of the repertiore is represented. This is used to calculate “Cut off greater than” count.</p>
<a class="reference internal image-reference" href="_images/4_TCRGEX_expanded.png"><img alt="Alternative text" src="_images/4_TCRGEX_expanded.png" style="width: 300px;" /></a>
I have also used this section in the STEGO.R V3 <a href="https://www.biorxiv.org/content/10.1101/2023.09.27.559702v3" target="_blank">preprint</a> for compairing the before and after treatment of a single clone.</section>
<section id="clustcr2">
<h3>ClusTCR2<a class="headerlink" href="#clustcr2" title="Permalink to this heading"></a></h3>
<p>This section will look into the clusters produced in STEP 2.</p>
<p>The user can select which chain they wish to interrogate e.g., TRA, TRB, TRG and TRD, as well as which cluster to interrogate.</p>
<p>The clsuters are ordered in how many total TCR were assocaited with a specific cluster.</p>
<p>The only additional plot that the user can interrogate is the motif.</p>
<a class="reference internal image-reference" href="_images/4_TCRGEX_clustering.png"><img alt="Alternative text" src="_images/4_TCRGEX_clustering.png" style="width: 300px;" /></a>
<a class="reference internal image-reference" href="_images/4_TCRGEX_motif.png"><img alt="Alternative text" src="_images/4_TCRGEX_motif.png" style="width: 1200px;" /></a>
</section>
<section id="epitope-tcrex">
<h3>Epitope (TCRex)<a class="headerlink" href="#epitope-tcrex" title="Permalink to this heading"></a></h3>
<p>This epitope section was designed to interrogate the TCRex outputs in STEGO.R.</p>
<p>This section includes  in the sidebar to add the TCR headers to the meta-data including the epitope and pathology.</p>
<p>The epitope list is ordered based on TCR with the most sequences.</p>
<p>The top associated epitope for the colitis compliation to melanoma treatment was the melanoma epitope ELAGIGILTV.</p>
<a class="reference internal image-reference" href="_images/4_TCRGEX_epitope.png"><img alt="Alternative text" src="_images/4_TCRGEX_epitope.png" style="width: 300px;" /></a>
<a class="reference internal image-reference" href="_images/4_TCRGEX_epitope_stats.png"><img alt="Alternative text" src="_images/4_TCRGEX_epitope_stats.png" style="width: 1200px;" /></a>
</section>
</section>
<section id="automation-tcr-gex">
<h2>Automation (TCR -&gt; GEX)<a class="headerlink" href="#automation-tcr-gex" title="Permalink to this heading"></a></h2>
<ul class="simple">
<li><dl class="simple">
<dt>Clonotype</dt><dd><ul>
<li><p>Download public (bar graphs), summary table, dot plot for each public-like clone</p></li>
<li><p>Download private (single sample)</p></li>
<li><p>Can restrict based on the calculated priority 1/(sample total * total count)</p></li>
</ul>
</dd>
</dl>
</li>
<li><dl class="simple">
<dt>Cluster</dt><dd><ul>
<li><dl class="simple">
<dt>Download the motif, summary table, dot plot for each public-like and private clusters</dt><dd><ul>
<li><p>A common cluster is TRAV1-2 TRAJ33</p></li>
<li><p>separate alpha and beta cut-offs</p></li>
</ul>
</dd>
</dl>
</li>
<li><p>Priority: 1/(number of nodes * sample total * total count)</p></li>
</ul>
</dd>
</dl>
</li>
<li><dl class="simple">
<dt>Epitope/Annotation</dt><dd><ul>
<li><p>with the epitopes find the associated epitopes from TCRex</p></li>
<li><p>Unselect “Add in Epitope data” to focus on the annotations. So, you can identify the TCR linked to specific annotation models (e.g., FunctionTcell)</p></li>
</ul>
</dd>
</dl>
</li>
</ul>
</section>
<section id="gex-tcr">
<h2>GEx -&gt; TCR<a class="headerlink" href="#gex-tcr" title="Permalink to this heading"></a></h2>
<ul class="simple">
<li><dl class="simple">
<dt>Annotation</dt><dd><ul>
<li><p>No longer being developed.</p></li>
</ul>
</dd>
</dl>
</li>
<li><dl class="simple">
<dt>Marker</dt><dd><ul>
<li><p>Single marker</p></li>
<li><p>Dual marker</p></li>
</ul>
</dd>
</dl>
</li>
</ul>
</section>
</section>


           </div>
          </div>
          <footer><div class="rst-footer-buttons" role="navigation" aria-label="Footer">
        <a href="3.Analysis.html" class="btn btn-neutral float-left" title="Step 3. Quality control" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
        <a href="5.UserTips.html" class="btn btn-neutral float-right" title="User Tips for the analysis" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2021, Graziella.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>